signer/WebCryptoSigner.js

```diff
@@ -0,0 +1,138 @@
+const jsrsasign = require('jsrsasign');
+const keyutil = require('js-crypto-key-utils');
+
+/**
+ * The signer based on Web Crypto API
+ */
+class WebCryptoSigner {
+  /**
+   * @param {String} pkcs1 PKCS#1
+   */
+  constructor(pkcs1) {
+    this.pkcs1 = pkcs1;
+  }
+
+  /**
+   * @param {Uint8Array} content
+   * @return {Uint8Array}
+   */
+  async sign(content) {
+    if (!this.pkcs8) {
+      this.pkcs8 = await this._PKCS1ToPKCS8(this.pkcs1);
+    }
+
+    const algorithm = { // EcdsaParams
+      name: 'ECDSA',
+      hash: 'SHA-256',
+    };
+    const data = content;
+    let key;
+
+    try {
+      key = await this._key(this.pkcs8);
+    } catch (_) {
+      throw new Error('Failed to sign the request');
+    }
+
+    try {
+      const signature = await window.crypto.subtle.sign(algorithm, key, data);
+      return this._P1363ToDer(new Uint8Array(signature));
+    } catch (_) {
+      throw new Error(`Failed to sign the request`);
+    }
+  }
+
+  /**
+   * @param {String} pkcs1
+   */
+  async _PKCS1ToPKCS8(pkcs1) {
+    pkcs1 = pkcs1.replace('-----BEGIN EC PRIVATE KEY-----', '')
+        .replace('-----END EC PRIVATE KEY-----', '')
+        .replace(/\n/g, '');
+    const k = jsrsasign.KEYUTIL.getKey(
+        jsrsasign.b64utohex(pkcs1), null, 'pkcs5prv'
+    );
+    const kk = new keyutil.Key('oct',
+        new Uint8Array(jsrsasign.hextoArrayBuffer(k.prvKeyHex)),
+        {namedCurve: 'P-256'}
+    );
+    return await kk.export('pem');
+  }
+
+  /**
+   * @param {Uint8Array} sig
```

Comment:
For the params also.
https://github.com/tokenio/sdk-js/blob/master/app/src/security/CryptoBrowser.js#L205